cobra/flux_analysis/sampling.py

```diff
@@ -84,14 +98,37 @@ class HRSampler(object):
     """
 
     def __init__(self, model, thinning, seed=None):
-        self.model = model
+        # This currently has to be done to reset the solver basis which is
+        # required to get deterministic warmup point generation
+        # (in turn required for a working `seed` arg)
+        if model.solver.is_integer:
+            raise TypeError("sampling does not work with integer problems :(")
+        self.model = model.copy()
         self.thinning = thinning
         self.n_samples = 0
-        self.S = create_stoichiometric_matrix(model, array_type='dense')
+        prob = constraint_matrices(model)
+        self.S = prob.equalities
```

Comment:
I agree. Do you think I should encapsulate all the matrices into a struct like object (namedtuple for instance)? So you would have something like sampler.problem.stoichiometry, sampler.problem.constraints, etc... Or rather leave it as sampler.stoichiometry?